### ISSUE TITLE
CodeSystem/$lookup returns confusing error when lookup did not resolve a code

### ISSUE THREAD

**Issue by csandersdev** [IBM/FHIR#1626]
**Describe the bug**
Using the CodeSystem/$lookup operation, if you provide it bad input data (i.e. an incorrect system value or a non-existent code), the system throws a 500 error and provides the response below. The [FHIR spec](https://www.hl7.org/fhir/codesystem-operation-lookup.html) says it should throw a 400 and an error message saying something like "code not found."

Current error:
```
{
    "resourceType": "OperationOutcome",
    "id": "ac-1e-c4-32-30de0912-ddd9-4753-a969-33a7a72a42df",
    "issue": [
        {
            "severity": "fatal",
            "code": "exception",
            "details": {
                "text": "FHIROperationException: An error occurred during the CodeSystem lookup operation"
            }
        }
    ]
}
```

**To Reproduce**
1. Create a codesystem using the provided bundle ([codesystem.txt](https://github.com/IBM/FHIR/files/5439377/codesystem.txt))
2. Run  CodeSystem/$lookup?system=http://hl7.org/fhir/CodeSystem/example&code=chol' to verify that the lookup operation works as intended.
3. Run  CodeSystem/$lookup?system=LOINC&code=chol' and note that you get an HTTP 500 with the error message above. 

**Expected behavior**
When a code cannot be resolved for the provided input parameters, the system should return an HTTP 400 with an appropriate "code not found" error message.

**Additional context**
Looking at the code for the [LookupOperation](https://github.com/IBM/FHIR/blob/master/operation/fhir-operation-term/src/main/java/com/ibm/fhir/operation/term/LookupOperation.java#L38) we are throwing an Exception on null data which gets immediately caught and then thrown again in a more generic error message. At minimum, the error message in the client would be better if the outer catch didn't obscure the error message.

**Comment by prb112** [comment]
In the doInvoke of the LookupOperation it has a general error as the caught exception. 

```
    @Override
    protected Parameters doInvoke(
            FHIROperationContext operationContext,
            Class<? extends Resource> resourceType,
            String logicalId,
            String versionId,
            Parameters parameters,
            FHIRResourceHelpers resourceHelper) throws FHIROperationException {
        try {
            Coding coding = getCoding(parameters, "coding", "code");
            LookupOutcome outcome = service.lookup(coding, LookupParameters.from(parameters));
            if (outcome == null) {
                throw new FHIROperationException("Lookup cannot be performed");
            }
            return outcome.toParameters();
        } catch (Exception e) {
            throw new FHIROperationException("An error occurred during the CodeSystem lookup operation", e);
        }
    }
}
```

In this case, the throw new exception would need to be mutated to a 400 response. 

We'd certainly consider a PR from a contributor on this.

**Comment by lmsurpre** [comment]
We believe this was addressed with PR #1642.  Please confirm.

**Comment by csandersdev** [comment]
I agree. Addressed with #1642.
